src/test/java/seedu/address/logic/commands/UntagCommandTest.java

```diff
@@ -0,0 +1,197 @@
+package seedu.address.logic.commands;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_TAG_FRIEND;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_TAG_HUSBAND;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_TAG_STUDENT;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_TAG_TEACHING_ASSISTANT;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_TAG_WIFE;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandFailure;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static seedu.address.logic.commands.CommandTestUtil.showPersonAtIndex;
+import static seedu.address.testutil.TypicalIndexes.INDEX_FIRST_PERSON;
+import static seedu.address.testutil.TypicalIndexes.INDEX_SECOND_PERSON;
+import static seedu.address.testutil.TypicalIndexes.INDEX_THIRD_PERSON;
+import static seedu.address.testutil.TypicalPersons.getTaggedTypicalAddressBook;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.model.AddressBook;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.person.Person;
+import seedu.address.model.tag.Tag;
+import seedu.address.testutil.EditPersonDescriptorBuilder;
+import seedu.address.testutil.PersonBuilder;
+
+class UntagCommandTest {
+
+    private Model generateFriendTaggedModel() {
+        return new ModelManager(getTaggedTypicalAddressBook(VALID_TAG_FRIEND), new UserPrefs());
+    }
+
+    private Model generateFriendAndHusbandTaggedModel() {
+        return new ModelManager(getTaggedTypicalAddressBook(VALID_TAG_FRIEND, VALID_TAG_HUSBAND), new UserPrefs());
+    }
+
+    private Model generateFriendHusbandTeachingAssistantStudentTaggedModel() {
+        return new ModelManager(getTaggedTypicalAddressBook(VALID_TAG_FRIEND, VALID_TAG_HUSBAND,
+                VALID_TAG_TEACHING_ASSISTANT, VALID_TAG_STUDENT), new UserPrefs());
+    }
+
+    @Test
+    void execute_singleTagNoRemainingTagsUnfilteredList_success() {
+        Model model = generateFriendTaggedModel();
+        Person firstPerson = model.getFilteredPersonList().get(INDEX_FIRST_PERSON.getZeroBased());
+
+        PersonBuilder personInList = new PersonBuilder(firstPerson);
+        Person editedPerson = personInList.withTags().build();
+
+        UntagCommand.EditPersonDescriptor descriptor = new EditPersonDescriptorBuilder()
```

Comment:
Maybe its better to just import the EditPersonDescriptor directly rather than typing UnTagCommand all the time?